File: mk1/mk1/Lista3/Zad2/Romb.cpp
#include "Romb.hpp"

Romb::Romb(int a, int kat) noexcept(false) {
    for (int i = 0; i < 4; i++) bok.push_back(a);
    this->kat = kat;
}

std::string Romb::Nazwa() noexcept(true) {
    return "Romb";
}

File: mk1/mk1/Lista3/Zad1/Romb.java
class Romb extends Czworokat{
    Romb(float a, float b, float c, float d, float alf){
        this.bok1 = a;
        this.bok2 = b;
        this.bok3 = c;
        this.bok4 = d;
        this.kat = alf;
    }

    public float Pole(){
        return (float) (bok1*bok2*Math.sin(kat));
    }

    public float Obwod(){
        return bok1+bok2+bok3+bok4;
    }

    public String Nazwa(){
        return "Romb";
    }
}